modules/services/nebula.nix

```
@@ -0,0 +1,265 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    filterAttrs
+    literalExpression
+    mapAttrsToList
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    recursiveUpdate
+    warnIf
+    types
+    ;
+
+  cfg = config.services.nebula;
+  enabledNetworks = filterAttrs (n: v: v.enable) cfg.networks;
+
+  format = pkgs.formats.yaml { };
+
+  nameToId = netName: "nebula-${netName}";
+
+  resolveFinalPort =
+    netCfg:
+    if netCfg.listen.port == null then
+      if (netCfg.isLighthouse || netCfg.isRelay) then 4242 else 0
+    else
+      netCfg.listen.port;
+
+in
+{
+  # Interface
+
+  options.services.nebula = {
+    package = mkPackageOption pkgs "nebula" { };
+    networks = mkOption {
+      description = "Nebula network definitions.";
+      default = { };
+      type = types.attrsOf (
+        types.submodule {
+          options = {
+            enable = mkOption {
```

```suggestion
            enable = mkEnableOption {
```

Can you replace all the booleans that default to `false` and just enable logic with `mkEnableOption`?

---

Implementing this change yields errors like:

```
error: cannot coerce a set to a string: { type = «thunk»; description = "Whether this node is a lighthouse."; }
```

I've reverted this for now, as it seems to be a stylistic rather than operational point - and I am using this in prod.

If there are in fact compelling reasons for this change, which I'm not tracking with, please help point me in the right direction 🙏